components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1260,9 +1260,58 @@ where
             SignificantMsg::RaftlogFetched { context, res } => {
                 self.on_raft_log_fetched(context, res);
             }
+            SignificantMsg::EnterForceLeaderState => self.on_enter_force_leader(),
+            SignificantMsg::ExitForceLeaderState => self.on_exit_force_leader(),
         }
     }
 
+    fn on_enter_force_leader(&mut self) {
+        info!(
+            "enter force leader state";
+            "region_id" => self.fsm.region_id(),
+            "peer_id" => self.fsm.peer_id(),
+        );
+        self.fsm.peer.force_leader = true;
+
+        if !self.fsm.peer.is_leader() {
+            // become candidate first to increase term
+            self.fsm.peer.raft_group.raft.become_candidate();
+            // trigger candidate twice to increase term by 2, to avoid there is
```

Comment:
What if there is a leader whose term is more than 3?

Reply:
it's never enough. And considering we use prevote, increasing term by 2 avoids most of the accident.

Reply:
How about making it campaign instead of becoming leader directly? If PD chooses a node with latest logs, it should be able to collect all remaining votes. If all remaining votes are collected, we force it as leader.